tests/hooks/useSynchronizedRequest.test.tsx

```diff
@@ -0,0 +1,26 @@
+import { useSynchronizedRequest } from '../../src/hooks/useSynchronizedRequest';
+import { renderHook } from '@testing-library/react-hooks';
+import { waitFor } from '@testing-library/react';
+
+
+let requestFunction = async () => 0;
+const errorFunction = () => 'error';
+
+it('returns an updated function with the same reference', async () => {
```

Comment:
I know I said one test, but looking at it again maybe we could also repeat this test for errorFunction?

Reply:
looking back, was there any benefit to using useRef on handleRejectedPromise?  (i.e. error function stuff) since it's not returned by `useSynchronizedRequest`?

Reply:
Done!  I made a test that checks: if you replace the error function and re-render, then the error function will change, but the `executeFunction` returned will be the same.